packages/mdc-list/foundation.js

```diff
@@ -291,6 +304,20 @@ class MDCListFoundation extends MDCFoundation {
       this.adapter_.focusItemAtIndex(lastIndex);
     }
   }
+
+  /**
+   * Toggles checkbox or radio at give index. Radio doesn't change the checked state if it is already checked.
+   * @param {number} index
+   */
+  toggleCheckboxOrRadioAtIndex_(index) {
+    if (!this.adapter_.hasCheckboxOrRadioAtIndex(index)) return;
+
+    let isChecked = true;
+    if (this.adapter_.hasCheckboxAtIndex(index)) {
+      isChecked = !this.adapter_.isCheckboxCheckedAtIndex(index);
+    }
+    this.adapter_.setCheckedCheckboxOrRadioAtIndex(index, isChecked);
```

Comment:
To clarify, was this adapter API changed primarily to move some of the logic into the foundation?